src/helpers/get-issues-data.ts

```diff
@@ -0,0 +1,138 @@
+import { FETCH_RANGED_COMMENTS } from "@/graphql/queries"
+import {
+    IssueCommentDataType,
+    IssueCommentNodes,
+    IssueComments
+} from "@/types/comments"
+
+export async function getIssueCommentsData({
+    username,
+    token,
+    query,
+    startDate,
+    endCursor
+}: {
+    username: string
+    token?: string
+    query: "FETCH_RANGED_COMMENTS"
+    startDate?: string
+    endCursor: string | null
+}) {
+    const tokenFromEnv = process.env.GITHUB_TOKEN
+
+    let graphqlQuery = ""
+    switch (query) {
+        case "FETCH_RANGED_COMMENTS":
+            graphqlQuery = FETCH_RANGED_COMMENTS
+            break
+        default:
+            throw new Error("Invalid query")
+    }
+
+    let hasNextPage = true
+
+    endCursor = startDate?.slice(0, 4) === "2022" ? endCursor : null
```

Comment:
we should fetch the current year, do a backdate by one year and use that here so we don't have to change years every year